lib/spack/spack/build_systems/autotools.py

```diff
@@ -187,7 +187,11 @@ def _removed_la_files_log(self):
     @property
     def archive_files(self):
         """Files to archive for packages based on autotools"""
-        files = [os.path.join(self.build_directory, "config.log")]
+        archives = ("config.log", "config.guess", "configure")
+        staging_prefixes = (self.build_directory, self.pkg.stage.source_path)
```

Comment:
Why do we have to check both? Won't the log always be in the same directory?

Reply:
In principle we shouldn't have to for config.log, if build_directory always points to the directory that's actually configured, but usually configure and config.guess will live only in the source tree, and sometimes config.log seems to elude us only looking in the build directory.